plugins/nf-azure/src/main/nextflow/cloud/azure/config/AzRegistryOpts.groovy

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2021, Microsoft Corp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package nextflow.cloud.azure.config
+
+
+import groovy.transform.CompileStatic
+
+/**
+ * Model Azure Batch registry config settings from nextflow config file
+ *
+ * @author Manuele Simi <manuele.simi@gmail.com>
+ */
+@CompileStatic
+class AzRegistryOpts {
+
+    private Map<String,String> sysEnv
+
+    String server
+    String userName
+    String password
+
+    AzRegistryOpts() {
+        this(Collections.emptyMap())
+    }
+
+    AzRegistryOpts(Map config, Map<String,String> env=null) {
+        assert config!=null
+        this.sysEnv = env==null ? new HashMap<String,String>(System.getenv()) : env
+        this.server = config.server ?: 'docker.io'
+        this.userName = config.userName ?: sysEnv.get('AZURE_REGISTRY_USER_NAME')
+        this.password = config.password ?: sysEnv.get('AZURE_REGISTRY_PASSWORD')
+    }
+
+    boolean configured() {
+        return this.userName || this.password
```

Comment:
Should not be `&&` ?

Reply:
No, I exactly meant `||`. This is intended to check whether the user configured "something" in the `registry` section. If one of the two is missing, then no private container registry is attached to the pool in the `AzBatchService` class. I believe is more sound to report the wrong configuration at its first usage. I thought this is also the case in other parts of NF.

If you think otherwise, I can have the `and`  condition here, and change the checks in the `AzBatchService`. But it wouldn't change much in my opinion.

Reply:
Then, maybe `configured` can be used to validate the required creds or throw an exception if incomplete in place of the following one 

https://github.com/nextflow-io/nextflow/pull/2078/files#diff-c2e05caf80270e25bb4207c82207f6530ad6b167924f91e86a021ec01e44ebf4R559

Reply:
Yep, that was the change I was testing. See the latest commits.